### PR TITLE
Switch package.json files to tabs

### PR DESCRIPTION
**Pull request by simison**
Spaces in `package.json` seem to be there for historic reasons; NPM enforced them until NPM v5 and [nowadays NPM doesn't care](https://github.com/npm/npm/issues/5694#issuecomment-307142476).

We've had a mixture or tabs and spaces in these files and for consistency, it's better if everything is tabs. 

This came up at https://github.com/Automattic/wp-calypso/pull/31542

I checked that there wasn't `.npmrc` config rule to enforce tabs always. :-(

See changes also in "Hide whitespace changes"  mode to see other types of changes done by Prettier.

#### In this PR:

- Switched root (https://github.com/Automattic/wp-calypso/commit/a4c632e70288d586b49fda019c612e874a277bc9) and sub-folder (https://github.com/Automattic/wp-calypso/commit/b751a53a7fb6d7d3e81702407109fa722e5454c2) package.json files to use tabs
   ```
    npx prettier \
      ./**/npm-shrinkwrap.json \
      ./**/package-lock.json \
      ./**/package.json \
      --write
   ```
- Removed editorconfig rule (https://github.com/Automattic/wp-calypso/commit/1e2e94ee4436d9688817c1d1e477d4a165d6d895) — left `*.yml` rule there:
    https://github.com/Automattic/wp-calypso/blob/2b2ca2b01466b33add52c533cb6383cef7080be2/.editorconfig#L20-L22
- Updated monorepo doc (https://github.com/Automattic/wp-calypso/commit/4f8a306a8113f197ad06e113eed086e655e74542, indentation & typo)

#### Testing

Dunno, make a cup of tea and run `npm run distclean && npm ci` ?

### CODE REVIEW

**Comment by simison** [comment]
Alternatively, Pretty with spaces: https://github.com/Automattic/wp-calypso/pull/31545

**Comment by sirreal** [comment]
I was encouraged to use spaces recently, although I can't find it for reference.

I'd prefer #31545

**Comment by simison** [comment]
I see wrong colours in Jetpack sidebar toggle logo but looks like it's unrelated to this PR and it's coming from changes in Gutenberg styles:

Screenshot 2019-03-19 at 12 20 37
Screenshot 2019-03-19 at 12 20 26

**Comment by simison** [comment]
Actually colours seme fine in `master`:

image

It's from here:

https://github.com/Automattic/wp-calypso/blob/bd2c8ac6d0c223f5fdc5aaa4071db86c3fa98916/packages/jetpack-blocks/src/shared/jetpack-plugin-sidebar.scss#L31-L39

In `master` we get this CSS:

```css
.edit-post-pinned-plugins .components-icon-button:not(.is-toggled) .jetpack-logo__icon-triangle {
    fill: #fff !important;
}
```

In this branch following:
```css
.edit-post-pinned-plugins .components-icon-button:hover .jetpack-logo__icon-triangle, .edit-post-pinned-plugins .components-icon-button:not(.is-toggled) .jetpack-logo__icon-triangle {
    fill: #fff !important;
    fill: var(--color-white) !important;
}
```

We rely quite a bit on these `--color-*` variables in our blocks so I'd expect other places to be broken as well.

File: docs/monorepo.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Publishing Packages with the Monorepo
 =====================================
 
-Caplyso is a monorepo. In addition to the Calypso application, it also hosts a number of independent modules that are published to NPM.
+Calypso is a monorepo. In addition to the Calypso application, it also hosts a number of independent modules that are published to NPM.
```
Comment (blowery) [inline docs/monorepo.md]: 😆

**Review by blowery — APPROVED** [review]
This seems fine to me. 

@simison have fun being on the blame line for all this stuff. 😆

**Comment by simison** [comment]
@sirreal any specific reason to prefer spaces over this one or shall I merge? :-)

**Comment by blowery** [comment]
@simison huh. That looks like the postcss custom properties plugin isn't running with the correct flag. How are you building that code?

**Comment by simison** [comment]
>  That looks like the postcss custom properties plugin isn't running with the correct flag. How are you building that code?

Woot, I'm in the wrong tab commenting on the wrong PR. :-) I'll move two of my comments over to https://github.com/Automattic/wp-calypso/pull/31503. Sorry noise!

**Comment by blowery** [comment]
> I was encouraged to use spaces recently, although I can't find it for reference.

@sirreal That was me. It was mostly just to keep the diffs small and keep annotations closer to who was actually responsible. But it seems like this is causing folks pain and folks can always run `git blame -w`.

**Comment by sirreal** [comment]
I don't care too much, but let's get prettier involved and hopefully the tabs/spaces thing is settled.

Tabs everywhere probably makes more sense given the rest of our codebase.

Let's add package.json files to reformat-files like #31545 did:

https://github.com/Automattic/wp-calypso/blob/90e501fc7f7c168e6b5d8e473b23a717981a10e3/package.json#L258

We could also autoformat package.json on precommit:

https://github.com/Automattic/wp-calypso/blob/ee560f15113df345029dc2a6bc05b951bf1bf109/bin/pre-commit-hook.js#L26-L31

**Comment by simison** [comment]
I did a quick test and Eslint wasn't happy with json files in commits.

Can't really add them to `.eslintignore` either because then Pretteir would ignore them. 🤔

**Comment by sirreal** [comment]
Prettier package.json on precommit and reformat https://github.com/Automattic/wp-calypso/pull/31554

**Pull request merged**